internal/backend/grpc/router/router.go

```diff
@@ -173,10 +173,28 @@ func (r *Router) Director(ctx context.Context, fullMethodName string) (proxy.Mod
 func (r *Router) getTalosBackend(ctx context.Context, md metadata.MD) ([]proxy.Backend, error) {
 	clusterName := getClusterName(md)
 
+	nodeID := getNodeID(md)
```

Comment:
it's not clear, does it work only with machine UUIDs?

Reply:
that's a great question. I'm afraid that with the hostname the resolution might be ambiguous.
I think I can make it work: we can show the error if the resolution gives several clusters.